### ISSUE TITLE
Teaching GitHub + Jekyll

### ISSUE THREAD

**Issue by thomasjbradley** [education/teachers#29]
I like to teach my students ideas that Jekyll provides: layouts, reusable code, separation, etc. but for the students I teach (Graphic Design students) as soon as we hit the command line all learning comes to a screeching halt. The mysterious, empty, black box gets in the way of learning Jekyll.

Now, I know that maybe it’s the way I cover the material, I just wish there was a slightly easier way to use Jekyll.

My thought is that since Jekyll is a critical aspect of GitHub Pages, that maybe there should be some sort of GUI on/off button to start the server. Maybe it’s inside the GitHub App or maybe not, but I want to help my students learn the topics of layouts, reusability, without the command line being a *huge* stumbling factor. (If anybody knows of such an app that would be fantastic!)

For me, I’m not training “programmers”, I’m training “designers”—and I know that division is a little blurry, and some of my students become programmers, but for the vast majority of them writing code isn’t a passion, but a necessity. The command line will never be something that interests them or they’ll ever find useful. I want them to make large websites with Jekyll, but the only thing they need the command line for is to install Jekyll and turn it on & off—it feels unnecessary.

Any thoughts on the matter would be greatly appreciated!

**Comment by eduOS** [comment]
Yeah, I find it's really a huge stumbling factor to conquer Jekyll command line.

**Comment by maxgaudin** [comment]
We find Jekyll a tab bit cumbersome to teach at a beginner's level and we're teaching them to become programmers rather than designers. Jekyll is actually not necessary at all to use with Github pages. 

We made a tutorial where students build a basic website in just HTML and host it on Github pages. We find this is a great start because they're not only creating a site but uploading it to the web where they can continue to build upon it. Check out the tutorial [here](https://github.com/OperationSpark/first-website).

**Comment by thomasjbradley** [comment]
@maxgaudin Yeah, for the first 3 terms we don’t touch Jekyll and just push plain HTML sites up to GitHub Pages.

But eventually I want the students to know about includes, DRY, etc. and that’s when we introduce Jekyll.

**Comment by tanyagray** [comment]
Check out CloudCannon.com - it's a web-based code editor and hosting service which supports Jekyll, and you can link it up to your GitHub account so that it does automatic pushes.

There is a free version which hides the site preview behind a password.

The only downside I foresee is if you want students being conscious of their git commits.

I'm working with very young students (11 yrs +) and will be trialling CloudCannon with them later this year. It looks really promising for my use case.
